File: src/Python/pipeline_services/examples/example01_text_clean.py
from __future__ import annotations

from pipeline_services.core.pipeline import Pipeline, StepControl
from pipeline_services.examples.text_steps import normalize_whitespace, strip


def truncate_at_280(text_value: str, control: StepControl) -> str:
    text_string = str(text_value)
    if len(text_string) <= 280:
        return text_string
    control.short_circuit()
    return text_string[0:280]


def main() -> None:
    pipeline = Pipeline("example01_text_clean", True)
    pipeline.add_action(strip)
    pipeline.add_action(normalize_whitespace)
    pipeline.add_action_named("truncate", truncate_at_280)

    result = pipeline.execute("  Hello   World  ")
    print("output=", result.context)
    print("shortCircuited=", result.short_circuited)
    print("errors=", len(result.errors))


if __name__ == "__main__":
    main()

File: src/typescript/src/pipeline_services/examples/example01_text_clean.ts
import { Pipeline, shortCircuit } from "../../index.js";
import { normalize_whitespace, strip } from "./text_steps.js";

async function truncateAt280(textValue: string): Promise<string> {
  if (textValue.length <= 280) {
    return textValue;
  }
  shortCircuit();
  return textValue.slice(0, 280);
}

async function main(): Promise<void> {
  const pipeline = new Pipeline<string>("example01_text_clean", true)
    .addAction(strip)
    .addAction(normalize_whitespace)
    .addAction(truncateAt280, "truncate");

  const result = await pipeline.runDetailed("  Hello   World  ");
  // eslint-disable-next-line no-console
  console.log("output=", result.context);
  // eslint-disable-next-line no-console
  console.log("shortCircuited=", result.shortCircuited);
  // eslint-disable-next-line no-console
  console.log("errors=", result.errors.length);
}

void main();
